### ISSUE TITLE
[critical!] Shuffle mechanism optimization required

### ISSUE THREAD

**Issue by juliancwirko** [juliancwirko/elven-nft-minter-sc#6]
The shuffling mechanism requires some optimization or a different approach because there are problems with gas when there are a lot of minted tokens already. It works well initially, but when there are more and more tokens minted, the mint endpoint starts to fail.

**Comment by juliancwirko** [comment]
The bug was more problematic than I thought, I am not sure if it is solved because it needs more testing, but there is a lot going on on the development branch, also in the CLI repo, last commit: https://github.com/juliancwirko/elven-nft-minter-sc/commit/6ea10c4574de2c0a104de0168d4a9da18e48f99c

**Comment by juliancwirko** [comment]
The problem should be solved in the development branch. It will soon be merged into the main branch.

More about the solution:

I had to add additional transaction calls responsible for populating VecMapper with indexes based on the total amount of tokens to mint. It is necessary to split big amounts into a couple of transactions because there is a gas limit per transaction, and they are pretty heavy. This way, the shuffling mechanism is quite performant because we will be operating on VecMapper using RandomnessSource and swap_remove.

The CLI tool will automatically split the amount into the proper transactions and populate the VecMapper. If you don't use the CLI, there is a populateIndexes endpoint (owner only). You would need to split the total amount into transactions where the amount will be max 5000. The amount provided when deploying should equal the sum of all amounts from populate transactions. The whole operation should be done only once when preparing the Smart Contract.

**Comment by juliancwirko** [comment]
Released 1.1.0 of the smart contract. It requires some testing and updates in the docs (some are already there).
